File: 전예진/Greedy/prob1.py
def solution(n, lost, reserve):
    # 체육복을 도난당한 학생들 중에서 여벌 체육복을 가지고 있는 학생들에 대한 처리
    lost_set = set(lost) - set(reserve)
    reserve_set = set(reserve) - set(lost)

    # 여벌 체육복을 가지고 있는 학생들 중에서 앞번호나 뒷번호에게 빌려줄 수 있는지 확인하고 빌려줌
    for r in reserve_set:
        if r - 1 in lost_set:
            lost_set.remove(r - 1)
        elif r + 1 in lost_set:
            lost_set.remove(r + 1)

    answer = n - len(lost_set)
    
    return answer

File: 주다애/Greedy/prob1.java
import java.util.*;

class Solution {
    public int solution(int n, int[] lost, int[] reserve) {
        List<Integer> losts = new ArrayList<>();
        List<Integer> reserves = new ArrayList<>();
        
        for(int i : lost) {
            losts.add(i);
        }
        for(int i : reserve) {
            reserves.add(i);
        }
        
        Collections.sort(losts);
        Collections.sort(reserves);
        
        for(int i = 0 ; i < reserves.size(); i++) {
            // 도난당한 학생은 자기 체육복 입으면 됨
            if(losts.contains(reserves.get(i))) {
                losts.remove(new Integer(reserves.get(i)));
                continue;
            }
            if(losts.contains(reserves.get(i) - 1)) {
                // 도난당한 학생인지 체크
                if(reserves.contains(reserves.get(i) - 1)) {
                    reserves.remove(new Integer(reserves.get(i) - 1));
                }
                losts.remove(new Integer(reserves.get(i) - 1));
                continue;
            }
            if(losts.contains(reserves.get(i) + 1)) {
                // 도난당한 학생인지 체크
                if(reserves.contains(reserves.get(i) + 1)) {
                    reserves.remove(new Integer(reserves.get(i) + 1));
                }
                losts.remove(new Integer(reserves.get(i) + 1));
                continue;
            }
        }
        
        int size = losts.size();
        return n - size;
    }
}
